lang/rs/wireguard/patch/0004-Fix-build-issues-for-benchmark-tests.patch

```diff
@@ -0,0 +1,183 @@
+diff --git a/src/platform/linux/cndp/packet.rs b/src/platform/linux/cndp/packet.rs
+index 6f3778f..9fc206e 100644
+--- a/src/platform/linux/cndp/packet.rs
++++ b/src/platform/linux/cndp/packet.rs
+@@ -627,7 +627,7 @@ mod tests {
+             &udp_payload[..],
+         );
+         // Parse the ethernet packet and verify the data.
+-        let ret = cndp_eth_packet.parse_eth_udp_packet();
++        let ret = cndp_eth_packet.parse_eth_udp_tcp_packet();
+         // Check if parsing succeeds.
+         assert_eq!(ret, 0);
+         // Check src, dst mac.
+diff --git a/src/wireguard/router/device.rs b/src/wireguard/router/device.rs
+index f3f0396..f74d25f 100644
+--- a/src/wireguard/router/device.rs
++++ b/src/wireguard/router/device.rs
+@@ -124,11 +124,15 @@ impl<E: Endpoint, C: Callbacks, T: tun::Writer, B: udp::Writer<E>> DeviceHandle<
+         let mut core_id: usize = 0;
+         while let Some(rx) = consumers.pop() {
+             threads.push(thread::spawn(move || {
+-                // Set core affinity for worker threads starting with worker_thread_base_coreid.
+-                if let Some(worker_thread_base_coreid) = worker_thread_base_coreid {
+-                    core_affinity::set_for_current(core_affinity::CoreId {
+-                        id: core_id + worker_thread_base_coreid,
+-                    });
++                cfg_if::cfg_if! {
++                    if #[cfg(all(target_os = "linux", feature = "cndp"))] {
++                        // Set core affinity for worker threads starting with worker_thread_base_coreid.
++                        if let Some(worker_thread_base_coreid) = worker_thread_base_coreid {
++                            core_affinity::set_for_current(core_affinity::CoreId {
++                                id: core_id + worker_thread_base_coreid,
++                            });
++                        }
++                    }
+                 }
+                 worker(rx);
+             }));
+@@ -201,7 +205,6 @@ impl<E: Endpoint, C: Callbacks, T: tun::Writer, B: udp::Writer<E>> DeviceHandle<
+ 
+         // ignore header prefix (for in-place transport message construction)
+         let packet = &msg[SIZE_MESSAGE_PREFIX..];
+-
+         let peer = self
+             .state
+             .table
+diff --git a/src/wireguard/router/tests/bench.rs b/src/wireguard/router/tests/bench.rs
+index c2334b3..579aa19 100644
+--- a/src/wireguard/router/tests/bench.rs
++++ b/src/wireguard/router/tests/bench.rs
+@@ -101,8 +101,10 @@ fn bench_router_outbound(b: &mut Bencher) {
+ 
+     // create device
+     let (_fake, _reader, tun_writer, _mtu) = dummy::TunTest::create(false);
++    // Use one worker thread for benchmark.
+     let router: Device<_, BencherCallbacks, dummy::TunWriter, dummy::VoidBind> =
+-        Device::new(num_cpus::get_physical(), tun_writer);
++        Device::new(1, tun_writer, None);
++    //Device::new(num_cpus::get_physical(), tun_writer, None);
```

Comment:
why is this commented out?

Reply:
I wanted to keep the original code commented to test with multiple CPUs as well. I missed adding a comment in code. I can create a new patch to add a comment rather than modifying existing patch. I noticed that some typos were fixed in existing patches earlier. But this somehow broke applying patches cleanly on top of wireguard public repo.